libs/dmlf/src/execution_result.cpp

```diff
@@ -0,0 +1,69 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "dmlf/execution/execution_result.hpp"
+
+namespace fetch {
+namespace dmlf {
+
+ExecutionResult::PromiseOfResult ExecutionResult::MakePromise()
+{
+  fetch::network::PromiseOf<ExecutionResult> promise{service::MakePromise()};
+  return promise;
+}
+
+void ExecutionResult::FulfillPromise(PromiseOfResult &promise, ExecutionResult const &fulfiller)
+{
+  serializers::MsgPackSerializer serializer;
```

Comment:
Not sure what "size" `fulfiller` might be here but there is a helper class that wraps a MsgPackSerializer and a counter together so that the result is pre-allocated. This seems to make sense to use that here *unless* you know that result is always going to be very small